packages/airnode-node/src/evm/verification/request-verification.ts

```diff
@@ -16,32 +16,30 @@ export function verifySponsorWallets<T>(
   requests: Request<T>[],
   masterHDNode: ethers.utils.HDNode
 ): LogsData<Request<T>[]> {
-  const logsWithVerifiedRequests: LogsData<Request<T>>[] = requests.map((request) => {
+  const logs: PendingLog[] = [];
+
+  const verifiedRequests = requests.filter((request) => {
     if (request.status !== RequestStatus.Pending) {
       const message = `Sponsor wallet verification skipped for Request:${request.id} as it has status:${request.status}`;
       const log = logger.pend('DEBUG', message);
-      return [[log], request];
+      logs.push(log);
```

Comment:
I can understand that `filter` is a bit easy to read than the current `map` implementation, but I don't think we should have side effects from a `.filter` callback.

Reply:
Agreed it's not ideal, but `map` no longer works, see [here](https://github.com/api3dao/airnode/pull/874#discussion_r815377836). I had a loop with two arrays, then `filter` was suggested so I went with that.